### PR TITLE
[#1793] TimeUtil: update DATE_FORMAT_REGEX

### LINKED ISSUE #1793: `extractDate()` unable to process `1/1/2022addedstring`
**Tell us about your environment**

* **RepoSense Version:** [`70b7185`](https://github.com/reposense/RepoSense/commit/70b71850c4ef2d6757ea993f69398a55a5244b78)
* **OS and Version:** Windows 11
* **Web Browser and Version (if applicable):**

**Please include the steps to reproduce the bug.**

1. Run `main` with `-s 1/1/2022addedstring` as input argument.

**What was expected to happen?**

As discussed [here](https://github.com/reposense/RepoSense/issues/1789), `1/1/2022addedstring` should also be accepted.

**What actually happened? Please include a screenshot of the output.**

Program exit with error as shown below

image

**If possible, include the URL to your RepoSense report or log files (if any).**

### PR DESCRIPTION
**Pull request by SkyBlaise99**
Fixes #1793

## Proposed commit message

```
The expected date input is 'dd/mm/yyyy' or 'd/m/yyyy'. With 
extractDate(), inputs like 'dd/mm/yyyyaddedstring' is also accepted.

The same behavior is expected with single-digit date input. However, 
extractDate() cannot process inputs such as '1/1/2022addedstring'.

Let's fix this by updating DATE_FORMAT_REGEX.
```

### CODE REVIEW

File: src/test/java/reposense/util/TimeUtilTest.java [hunk 1]
```diff
@@ -23,6 +23,29 @@ public void extractDate_validDateAndTime_success() {
...
+    public void extractDate_validSingleDigitDateAndTime_success() {
+        String originalDateAndTime = "1/1/2022 12:34:56";
+        String expectedDate = "1/1/2022";
+        String actualDate = TimeUtil.extractDate(originalDateAndTime);
+        Assertions.assertEquals(expectedDate, actualDate);
+    }
+
+    @Test
+    public void extractDate_validSingleDigitDateAndString_success() {
+        String originalDateAndTime = "1/1/2022 addedstring";
```
[the remark this answers is not in the document]
Comment (SkyBlaise99) [reply src/test/java/reposense/util/TimeUtilTest.java:43]: Thank for pointing that out, will update it quickly.

**Review by chan-j-d — APPROVED** [review]
Code LGTM! Good work!

Regarding the commit message, correct me if I'm wrong but I think the issue at hand is that 01/01/2012abcde works, 1/1/2012 works but 1/1/2012abcde does not. I think this context should be added to the commit message as currently, it does not exactly explain why we might want to include such behavior.

**Comment by SkyBlaise99** [comment]
> Regarding the commit message, correct me if I'm wrong but I think the issue at hand is that 01/01/2012abcde works, 1/1/2012 works but 1/1/2012abcde does not.

Thanks for the feedback, I'll think about how to include it in.

**Comment by SkyBlaise99** [comment]
Hi @dcshzj, can you help to review this PR? Thank you!

**Review by dcshzj — APPROVED** [review]
LGTM! Great catch.

**Pull request merged**